src/python/pants/option/global_options.py

```diff
@@ -284,8 +293,20 @@ def register_bootstrap_options(cls, register):
                   '(e.g. BUILD file scanning, glob matching, etc). '
                   'Patterns use the gitignore syntax (https://git-scm.com/docs/gitignore). '
                   'The `--pants-distdir` and `--pants-workdir` locations are inherently ignored.')
+    register("--files-not-found-behavior", advanced=True,
+             type=FileNotFoundBehavior, default=FileNotFoundBehavior.warn,
+             help="What to do when files specified on the command line or in BUILD files cannot be "
+                  "found. This happens when the files do not exist on your machine or when they "
+                  "are ignored by the `--pants-ignore` option.")
     register('--glob-expansion-failure', advanced=True,
-             default=GlobMatchErrorBehavior.warn, type=GlobMatchErrorBehavior,
+             type=GlobMatchErrorBehavior, default=GlobMatchErrorBehavior.warn,
+             removal_version="1.27.0.dev0",
+             removal_hint="If you currently set `--glob-expansion-failure=error`, instead set "
+                          "`--files-not-found-behavior=error`.\n\n"
+                          "If you currently set `--glob-expansion-failure=ignore`, you will "
+                          "need to instead either set `--files-not-found-behavior=warn` (the "
+                          "default) or `--files-not-found-behavior=error`. Ignoring when files are "
```

Comment:
`ignore` is important for large repositories migrating from one behavior to another. It's a good idea not to support it on the CLI, but I think that that just points to overloading one options value with two meanings.

Reply:
This PR doesn't remove support, it just deprecates it, no? Giving two months to clean up a repo sounds reasonable; would you be happier with a 3 or 4 version deprecation period, rather than two?

Reply:
Yes, I’m fine with a longer deprecation. 

I do think we want to stop allowing `ignore` in BUILD files too. It leads to subtle bugs.

Reply:
That is a smaller issue compared to the larger issue: I think that this option overloads something that should not be overloaded.

Reply:
Stu, ideas on what this would ideally be called? `--file-args-not-found-behavior` and `--target-files-not-found-behavior`? `--file-args-not-found-behavior` and `--sources-not-found-behavior`?

While I do think it should be a unified option, I'm not very strongly committed to that and will defer to your judgment here. I only don't know what to call the two things.

Reply:
I'm not sure we need an option to control the CLI behaviour... I'm guessing that we can agree on either warn or error for that case: I'm fine with either, although aligning with what happens when an address spec doesn't match something is useful.

Reply:
Ah, that's a good way around the problem. 

I'm fine with always warning for CLI behavior and not allowing you to configure that. Very opposed to erroring - as previously described, we can still do meaningful work when you run `./pants cloc src/python/util:: src/java/util::` and it turns out you no longer have a `src/java/util` folder. The warning is abundantly obvious and the user can rerun the command if they care to.

So, that leaves this option to being `--files-not-found-behavior` but only impacting BUILD file behavior, rather than file args. SGTM.

Reply:
> The warning is abundantly obvious and the user can rerun the command if they care to.

I'm pretty concerned about how this is exposed to scripts - an error is much more friendly in this case than a warning, where you may find that a command which used to do something (say "run my project's tests") now silently does nothing and succeeds.

Reply:
> I'm pretty concerned about how this is exposed to scripts - an error is much more friendly in this case than a warning, where you may find that a command which used to do something (say "run my project's tests") now silently does nothing and succeeds.

Oh, huh, true. Good point. This would also align with how we deal with address specs.

@illicitonion do you propose always erroring or adding an option to configure this? I do like the simplicity of Stu's proposal to not have an option. We don't allow an option for address specs, for example.

Reply:
I'd say always error, no option